pylib/spinnaker/yaml_util.py

```diff
@@ -186,13 +188,96 @@ def update_yml_source(path, update_dict):
     with open(path, 'r') as source_file:
       source = source_file.read()
       for prop in updated_keys:
-        source = bindings.transform_yaml_source(source, prop)
+        source = bindings.transform_yaml_source(source, prop,
+                                                add_new_nodes=add_new_nodes)
 
     with open(path, 'w') as source_file:
       source_file.write(source)
 
+  def find_yaml_context(self, source, root_node, full_key, raise_if_not_found):
+    """Given a raml file and full key, find the span of the value for the key.
 
-  def transform_yaml_source(self, source, key):
+    If the key does not exist, then return how we should modify the file
+    around the insertion point so that the context is there.
+
+    Args:
+      source: [string] The YAML source text.
+      root_node: [yaml.nodes.MappingNode]  The composed yaml tree
+      full_key: [string] Dot-delimited path whose value we're looking for
+      raise_if_not_found: [boolean] Whether to raise a KeyError or return
+         additional context if key isnt found.
+
+    Returns:
+      context, span
+
+      where:
+          context:[string, string)]: Text to append before and after the cut.
+          span: [(int, int)]: The start/end range of source with existing value
+                               to cut.
```

Comment:
Seems like this indentation should either back out one space or just have the 2 words on the previous line.